### ISSUE TITLE
Duplicate categories

### ISSUE THREAD

**Issue by timothymcmackin** [SenorTacoo/FibbageXLCustomQuestionsEditor#1]
Hi, and thanks for the awesome tool! I'm using it to put together a custom game for my neighborhood based on silly emails that we get to the neighborhood mailing list.

When I run it, I see duplicate categories. Maybe there is a limit to the number of questions in a category? Sometimes I see different duplicated categories.

Here is my fibbageshortie.jet file. The categories appear to be the same here and in the individual question files:

```
{"content":[{"x":false,"id":49733,"category":"Free on the curb","bumper":""},{"x":false,"id":61305,"category":"Want to buy","bumper":""},{"x":false,"id":12761,"category":"Free on the curb","bumper":""},{"x":false,"id":61248,"category":"Free on the curb","bumper":""},{"x":false,"id":19241,"category":"Want to buy","bumper":""},{"x":false,"id":26349,"category":"Just weird","bumper":""},{"x":false,"id":2938,"category":"Free on the curb","bumper":""},{"x":false,"id":14788,"category":"Free on the curb","bumper":""},{"x":false,"id":13407,"category":"Just weird","bumper":""},{"x":false,"id":1559,"category":"Want to buy","bumper":""},{"x":false,"id":58974,"category":"Want to buy","bumper":""},{"x":false,"id":6190,"category":"Just weird","bumper":""},{"x":false,"id":34121,"category":"Just weird","bumper":""},{"x":false,"id":5349,"category":"Just weird","bumper":""},{"x":false,"id":14684,"category":"Just weird","bumper":""},{"x":false,"id":11963,"category":"Free on the curb","bumper":""},{"x":false,"id":20122,"category":"Just weird","bumper":""},{"x":false,"id":40319,"category":"Just weird","bumper":""},{"x":false,"id":65058,"category":"Want to buy","bumper":""},{"x":false,"id":20999,"category":"Just weird","bumper":""},{"x":false,"id":65041,"category":"Free on the curb","bumper":""},{"x":false,"id":47512,"category":"Free on the curb","bumper":""},{"x":false,"id":31771,"category":"Just weird","bumper":""}],"episodeId":0}
```
![Screenshot_20220121-193023](https://user-images.githubusercontent.com/6494785/150616808-bb522062-5c75-4dce-b319-c254c422783f.png)

**Comment by SenorTacoo** [comment]
Hi @timothymcmackin, I need some more information about your problem:

1. Did you add a question with the same category as another question or did it appear by itself?
1.1. If it happened by itself, please provide some steps to reproduce the bug (I tried by myself but had no success).
1.2. If you added questions with the same category, I cannot help because that's how Fibbage works (like in original questions, every category is unique).

Thank you for using my application, I am glad that it can be useful to someone 😄.

**Comment by timothymcmackin** [comment]
ok, I will do some debugging when I can and provide you with more information. It's a great idea and I hope I can help. :-)

**Comment by timothymcmackin** [comment]
@SenorTacoo I did some experimenting and it seems that as I play the game, it duplicates different categories. For the first question, it might show category A twice and categoy B 3 times. Then for question 2, it shows category C twice and A and B only once. Strange.

I've also noticed that on my test runs, it always locks up after the 3rd or 4th question, and I get a blank screen. :-(

If it helps, here is my custom game:
https://github.com/timothymcmackin/BestOfListserv

**Comment by SenorTacoo** [comment]
I have checked your questions @timothymcmackin, and I got a blank screen a few times as well. I found out that you need at least _2*number_of_players+1_ suggestions (lie for me option + game's lie) for the game to not freeze randomly. Since you can have 8 players max, the best number of suggestions is 17. A minimum number of suggestions is 5 for a game of 2 players. 
When I added more suggestions to your questions, the game never crashed (hopefully that's the fix, try it yourself 😃).

One of the symptom of too few _Suggestions_ is the _NULL/empty_ lie option like in this picture (when I clicked lie for me).
![image](https://user-images.githubusercontent.com/74991570/160244668-68ac255b-48d6-4a3e-8176-7714f7191603.png)


For the duplicated categories - the game shows random questions based on their _ID_, not _Category_, so you will see questions with the same _Category_ but not with the same _ID_. Look at the original questions, every question has a different _Category_ and a lot of _Suggestions_.

![image](https://user-images.githubusercontent.com/74991570/160244791-0e125a29-031f-464c-a545-6ec77418c785.png)

I removed the answer to not spoil it to other players 😉 
![image](https://user-images.githubusercontent.com/74991570/160244875-15f8346a-81cf-4bbb-95c9-6d4de0897335.png)

In summary, try adding more _Suggestions_ and remember every question should have a unique _ID_ and _Category_.

I'm thinking about adding a message box with information about duplicated _Categories_ or too few _Suggestions_ while saving, so users will be aware that the game may freeze or it may show the same _Category_ during playing.

**Comment by timothymcmackin** [comment]
Ok, I will add enough suggestions. Do you know if I can repeat suggestions, or do I need enough unique suggestions for players * 2 + 1?

**Comment by timothymcmackin** [comment]
I added some suggestions and duplicated them so there were plenty of them and I was able to get through a game without crashes, so duplicates are probably ok. When I clicked "Lie for me" it did show the same answer twice, which is to be expected.

**Comment by SenorTacoo** [comment]
Duplicated suggestions are ok, but be aware that multiple players can have the same lie, and during "selecting answer" you can end up with one true answer and 3 the same lies.

**Comment by SenorTacoo** [comment]
I've created a new release (0.0,2) with information about duplicated categories and too few suggestions. I think that's it ok to close this issue for now.
